light-client/src/supervisor.rs

```diff
@@ -313,51 +314,58 @@ impl Supervisor {
 /// A handle to a `Supervisor` which allows to communicate with
 /// the supervisor across thread boundaries via message passing.
 pub struct Handle {
-    sender: channel::Sender<Event>,
+    sender: channel::Sender<HandleInput>,
 }
 
 impl Handle {
     /// Crate a new handle that sends events to the supervisor via
     /// the given channel. For internal use only.
-    pub fn new(sender: channel::Sender<Event>) -> Self {
+    fn new(sender: channel::Sender<HandleInput>) -> Self {
         Self { sender }
     }
 
+    /// Get latest trusted block from the [`Supervisor`].
+    pub fn latest_trusted(&mut self) -> Result<Option<LightBlock>, Error> {
```

Comment:
In what case does this return Ok(`None`)?

Reply:
When the underlying primary returns `None` for its `latest_trusted`. This is really a direct pass-through, and judging by the method on the Instance the latest trusted being none should be expected and is non-errornous behaviour.

Reply:
Makes sense. Need to check this out locally. I'm still wondering, when a primary doesn't yet have any trusted light block in store:
https://sourcegraph.com/github.com/informalsystems/tendermint-rs@b3a5b1e70cd2d7fb06361950d10951d51b3dce06/-/blob/light-client/src/supervisor.rs#L53-55

Reply:
If the primary always returns a trusted block, we should encode that in our types. While primary/witness might only be a operational concern as @milosevic pointed out during the call yesterday, there might a possibility to otherwise encode this expectation. For example with an enum distinguishing between node types/states. What would be the right way to find answers to these questions?

Reply:
This might be orthogonal to this PR but I think, we'd need to enumerate in which cases the primary does not yet have a trusted light block and then figure out if we want to treat these cases as errors for some use-cases instead (maybe further up, e.g. in the rpc).

Reply:
Follow-up in #395

Reply:
Left a comment in #395. Otherwise, I fully agree that we should try encode such invariants as possible in the types themselves.